parquet-hadoop/src/test/java/org/apache/parquet/statistics/RandomValueGenerators.java

```diff
@@ -0,0 +1,395 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.statistics;
+
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.PrimitiveType;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.math.BigInteger;
+import java.util.*;
+
+public class RandomValueGenerators {
+  private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
+
+  public static abstract class RandomValueGenerator<TValueType extends Comparable<TValueType>> implements Iterable<TValueType> {
+    public static abstract class ValueIterator<TValueType> implements Iterator<TValueType> {
+      private int count;
+
+      public ValueIterator(int count) {
+        this.count = count;
+      }
+
+      @Override
+      public void remove() {
+        throw new NotImplementedException();
+      }
+
+      @Override
+      public boolean hasNext() {
+        return (count --> 0);
+      }
+    }
+
+    protected final Random random;
+    protected final PrimitiveType.PrimitiveTypeName forType;
+    protected final int bufferLength;
+    protected final int total;
```

Comment:
Actually, I see what you mean. Since there is no maximum, there's no need for `total` either. I'll remove this.